File: backend/src/schemas/task.py
from pydantic import BaseModel
import uuid
from datetime import datetime


class TaskCreate(BaseModel):
    name: str
    description: str | None = None
    category: str | None = None
    priority: int = 3
    estimated_hours: float | None = None
    assigned_to: str | None = None
    dependency_ids: list[uuid.UUID] = []


class TaskUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: str | None = None
    priority: int | None = None
    estimated_hours: float | None = None
    actual_hours: float | None = None
    assigned_to: str | None = None


class TaskOut(BaseModel):
    id: uuid.UUID
    plan_id: uuid.UUID
    version: int
    name: str
    description: str | None
    category: str | None
    status: str
    priority: int
    estimated_hours: float | None
    actual_hours: float | None
    planned_start: datetime | None
    planned_end: datetime | None
    actual_start: datetime | None
    actual_end: datetime | None
    assigned_to: str | None
    is_on_critical_path: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class DependencyCreate(BaseModel):
    predecessor_id: uuid.UUID
    dep_type: str = "finish_to_start"
    lag_hours: float = 0.0

File: frontend/src/types/task.ts
export interface Task {
  id: string
  plan_id: string
  version: number
  name: string
  description?: string
  category?: string
  status: 'pending' | 'in_progress' | 'blocked' | 'completed' | 'skipped' | 'failed'
  priority: number
  estimated_hours?: number
  actual_hours?: number
  planned_start?: string
  planned_end?: string
  actual_start?: string
  actual_end?: string
  assigned_to?: string
  is_on_critical_path: boolean
  created_at: string
}

export interface TimelineEntry {
  task_id: string
  name: string
  category?: string
  status: string
  is_on_critical_path: boolean
  planned_start?: string
  planned_end?: string
  actual_start?: string
  actual_end?: string
  pct_complete: number
  is_delayed: boolean
}
